engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -729,6 +791,53 @@ namespace dmGameSystem
         }
     }
 
+    static void RenderListFrustumCulling(dmRender::RenderListVisibilityParams const &params)
+    {
+        DM_PROFILE("Mesh"); // TODO - use a proper name here
+
+        MeshWorld* mesh_world = (MeshWorld*)params.m_UserData;
+
+
+        const dmIntersection::Frustum frustum = *params.m_Frustum;
+        uint32_t num_entries = params.m_NumEntries;
+        for (uint32_t i = 0; i < num_entries; ++i)
+        {
+            dmRender::RenderListEntry* entry = &params.m_Entries[i];
+            MeshComponent* component_p = (MeshComponent*)entry->m_UserData;
+
+            dmGameSystem::BufferResource* br = GetVerticesBuffer(component_p, component_p->m_Resource);
+            dmVMath::Point3 boundsMin, boundsMax;
+            dmBuffer::Result r = dmBuffer::GetBounds(br->m_Buffer, boundsMin, boundsMax);
+            if (r != dmBuffer::RESULT_OK) {
+                dmLogWarning("Could not get culling bounds for mesh buffer. Is the buffer valid ?");
+                continue;
+            }
+
+            // are there any actual bounds set ? (checking if boundxMin still contains its initial values should be enough)
+            if (boundsMin.getX() == BOUNDS_MIN_EMPTY.getX() && boundsMin.getY() == BOUNDS_MIN_EMPTY.getY() && boundsMin.getZ() == BOUNDS_MIN_EMPTY.getZ()) {
+                continue;
+            }
+
+            // get center of bounding box in local coords
+            float center_x = (boundsMax.getX() + boundsMin.getX())/2;
+            float center_y = (boundsMax.getY() + boundsMin.getY())/2;
+            float center_z = (boundsMax.getZ() + boundsMin.getZ())/2;
```

Comment:
Doing this calculation each frame, I wonder if the data we stored is wrong?
Should we store "center+half_extent" instead?

Reply:
ok then. I'll keep that in mind.